src/openforms/submissions/api/viewsets.py

```diff
@@ -108,60 +102,64 @@ def _complete(self, request, *args, **kwargs):
+        TODO: give some leeway? add endpoint to trigger cleanup immediately?
+        TODO: emit a status URL with time-based token (Salted HMAC, from token generator)
+        that the frontend can poll without needing the submission in the session.
+        TODO: if appointment registration fails, the submission ID must be added back
+        to the session so the user can return to the first step(s) in the UI.
         """
         submission = self.get_object()
         validate_submission_completion(submission, request=request)
-        submission.completed_on = timezone.now()
 
+        submission.completed_on = timezone.now()
         submission.save()
+
+        # TODO: implement in celery tasks in cleanup (see ./tasks/__init__.py)
         remove_submission_from_session(submission, self.request.session)
         remove_submission_uploads_from_session(submission, self.request.session)
 
-        submission_report = SubmissionReport.objects.create(
-            title=_("%(title)s: Submission report") % {"title": submission.form.name},
-            submission=submission,
-        )
-
-        def on_submission_commit():
-            # The submission report needs to already have been generated before it can be attached to the zaak
-            # that is created in the registration
-            chain = generate_submission_report.si(
-                submission_report.id
-            ) | register_submission.si(submission.id)
-            chain.delay()
-
-            # this can run any time because they have been claimed earlier
-            cleanup_temporary_files_for.delay(submission.id)
+        # after committing the database transaction where the submissions completion is
+        # stored, start processing the completion.
+        transaction.on_commit(lambda: on_completion(submission.id))
 
-        transaction.on_commit(on_submission_commit)
+        return Response(status=status.HTTP_204_NO_CONTENT)
 
-        book_appointment_for_submission(submission)
-
-        if hasattr(submission.form, "confirmation_email_template"):
-            transaction.on_commit(lambda: send_confirmation_email(submission))
-
-        token = token_generator.make_token(submission_report)
-        download_report_url = reverse(
-            "api:submissions:download-submission",
-            kwargs={"report_id": submission_report.id, "token": token},
-        )
-        report_status_url = reverse(
-            "api:submissions:submission-report-status",
-            kwargs={"report_id": submission_report.id, "token": token},
-        )
+    @extend_schema(
+        summary=_("Get the submission processing status"),
+        request=None,
+        responses={
+            200: SubmissionProcessingStatusSerializer,
+            # TODO: 403, 401, 429
+        },
+    )
+    @action(
+        detail=True,
+        methods=["get"],
+        permission_classes=(ActiveSubmissionPermission,),
+    )
+    def status(self, request, *args, **kwargs):
```

Comment:
This is broken in its current state, but will be tackled in a follow up PR. The complete endpoint removes the submission ID from the session, so this permission class will always cause a 403.

Tests will be added for this endpoint in aforementioned follow up PR.